core/src/main/java/dev/morphia/mapping/MorphiaDefaultsConvention.java

```diff
@@ -149,6 +140,9 @@ private Handler getHandler(FieldModelBuilder builder) {
             if (handler == null) {
                 for (Annotation annotation : builder.annotations()) {
                     handler = annotation.annotationType().getAnnotation(Handler.class);
+                    if (handler != null) {
```

Comment:
I've chose this style to make it plain obvious.
Sure, this loop can be switched to C-style like with checking `handler == null` in the condition (but then it'll be IMHO less obvious) or conditionally assign the `handler` if `handler == null`, but this might also be more blurry. I guess a Stream with the `findFirst` is an overkill...
So what to do here?

Reply:
change to using an iterator and a while.  `while (iterator.hasNext() && handler == null)`

Reply:
Done. BTW: is there like a code-style guide for Morphia?

Reply:
There's a checkstyle guide that is based loosely, iirc, on the Sun guide before that sublimated from existence.  This style guide goes all the way back to my MongoDB days so I don't recall exactly its origins.  But nothing written, sadly.  It might be worth, at some point, revisiting that definition given that Morphia is now on Java 11 and not 6...